source/extensions/http/header_validators/envoy_default/http1_header_validator.cc

```diff
@@ -288,10 +282,17 @@ void Http1HeaderValidator::sanitizeContentLength(::Envoy::Http::RequestHeaderMap
         header_map.removeContentLength();
       }
     }
-  } else if (header_map.ContentLength() && header_map.getContentLengthValue() == "0" &&
-             ::Envoy::Http::HeaderUtility::isConnect(header_map)) {
+  }
```

Comment:
nit: instead of 3 ifs, can we use 1 here?

Reply:
done